editor.planx.uk/src/@planx/components/Pay/Editor/Editor.tsx

```diff
@@ -0,0 +1,185 @@
+      },
+    ],
+    ...parseBaseNodeData(props.node?.data),
+  };
+
+  const onSubmit = (newValues: Pay) => {
+    if (props.handleSubmit) {
+      props.handleSubmit({ type: TYPES.Pay, data: newValues });
+    }
+  };
+
+  return (
+    <Formik<Pay>
+      initialValues={initialValues}
+      onSubmit={onSubmit}
+      validationSchema={validationSchema}
+      validateOnChange={true}
+      validateOnBlur={true}
+    >
+      {({
+        values,
+        handleChange,
+        setFieldValue,
+      }) => (
+        <Form id="modal" name="modal">
+          <ModalSection>
+            <ModalSectionContent title="Payment" Icon={ICONS[TYPES.Pay]}>
+              <InputRow>
+                <Input
+                  format="large"
+                  placeholder="Page title"
+                  name="title"
+                  value={values.title}
+                  onChange={handleChange}
+                />
+              </InputRow>
+              <InputRow>
+                <Input
+                  format="bold"
+                  placeholder="Banner title"
+                  name="bannerTitle"
+                  value={values.bannerTitle}
+                  onChange={handleChange}
+                />
+              </InputRow>
+              <InputRow>
+                <RichTextInput
+                  placeholder="Banner description"
+                  name="description"
+                  value={values.description}
+                  onChange={handleChange}
+                />
+              </InputRow>
+              <InputRow>
+                <Input
+                  format="data"
+                  name="fn"
+                  value={values.fn}
+                  placeholder="Data field for calculated amount"
+                  onChange={handleChange}
+                />
+              </InputRow>
+            </ModalSectionContent>
+            <ModalSectionContent>
+              <InputRow>
+                <Input
+                  format="large"
+                  placeholder="Instructions title"
+                  name="instructionsTitle"
+                  value={values.instructionsTitle}
+                  onChange={handleChange}
+                />
+              </InputRow>
+              <InputRow>
+                <RichTextInput
+                  placeholder="Instructions description"
+                  name="instructionsDescription"
+                  value={values.instructionsDescription}
+                  onChange={handleChange}
+                />
+              </InputRow>
+              <InputRow>
+                <Switch
+                  checked={values.hidePay}
+                  onChange={() => setFieldValue("hidePay", !values.hidePay)}
+                  label="Hide the pay buttons and show fee for information only"
+                />
+              </InputRow>
+            </ModalSectionContent>
+          </ModalSection>
+          <GovPayMetadataSection/>
+          <InviteToPaySection/>
```

Comment:
These are now their own components, and not embedded within a single very large component.